src/runtime/GlobalObjectBuiltinRegExp.cpp

```diff
@@ -350,6 +350,135 @@ static Value builtinRegExpSplit(ExecutionState& state, Value thisValue, size_t a
     return A;
 }
 
+static Value builtinRegExpReplace(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    ASSERT(argv != nullptr);
+    RESOLVE_THIS_BINDING_TO_OBJECT(obj, RegExp, replace);
+    RegExpObject* rx = obj->asRegExpObject(state);
+    String* str = argv[0].toString(state);
+    ASSERT(str != nullptr);
+    Value replaceValue = argv[1];
+    size_t lengthStr = str->length();
+    bool functionalReplace = replaceValue.isCallable();
+    bool fullUnicode = false;
+    String* replacement = String::emptyString;
+    // 7. If functionalReplace is false, then
+    if (functionalReplace == false) {
+        replaceValue = replaceValue.toString(state);
+    }
+
+    bool global = rx->option() & RegExpObject::Option::Global;
+
+    // 10. If global is true, then
+    if (global == true) {
+        fullUnicode = rx->option() & RegExpObject::Option::Unicode;
+        rx->setLastIndex(state, Value(0));
+    }
+    ValueVector results;
+    // 13. Repeat, while done is false
+    while (true) {
+        Value result = builtinRegExpExec(state, obj, 1, &argv[0], false);
+        // 13.c If result is null, set done to true
+        if (result.isNull() == true) {
+            break;
+        }
+        // 13.d
+        results.pushBack(result);
+        // 13.d.ii. If global is false, set done to true.
+        if (global == false) {
+            break;
+        }
+
+        // 13.d.iii.
+        Value matchStr = result.asObject()->get(state, ObjectPropertyName(state, Value(0))).value(state, Value(0));
```

Comment:
`.value(state, Value(0))` should be changed to `.value(state, result)`, since whenever the property is an accessor, the function object need to be passed as the this value to invoke the getter function.